scripts/idl/generators/java/ChipClustersCpp.jinja

```diff
@@ -1,14 +1,16 @@
 {%- macro encode_optional(target, source, depth, encodable) -%}
-  if ({{source}} != nullptr) {
-    jobject optionalValue_{{depth}};
+  {
```

Comment:
could we add a unit test in https://github.com/project-chip/connectedhomeip/tree/master/scripts/idl/tests ?

I would like to understand the generated code and how it looks like.